timemachine/cpp/src/bd_exchange_move.cu

```diff
@@ -217,21 +218,39 @@ void BDExchangeMove<RealType>::move(
             d_lse_exp_sum_after_.data,
             stream);
 
-        k_attempt_exchange_move<RealType><<<1, 1, 0, stream>>>(
-            N,
-            d_mh_noise_.data + (step * batch_size_),
+        k_select_first_valid_move<RealType><<<1, min(512, batch_size_), 0, stream>>>(
```

Comment:
we should modify determinism tests above for batch_sizes: `{> 512, < 512, = 512}` to make sure this logic is correct

Reply:
Added test in https://github.com/proteneer/timemachine/pull/1274/commits/e8b45c58a356b3b7746445219c4ec00d841ae1c8